lib/index.js

```diff
@@ -53,7 +53,9 @@ class Oppsy extends Events.EventEmitter {
                     emit[taskName] = results.shift();
                 }
 
-                this.emit('ops', emit);
+                // Emit a deep copy so we can modify this data internall without
+                // affecting listeners.
+                this.emit('ops', Hoek.clone(emit));
```

Comment:
This isn't really a race condition, just pass by reference. Also, cloning is an expensive operation, wouldn't it be better to leave this up to the consumers?

Reply:
You are right, this is a pass-by-reference that makes a race condition very easy to happen between consumers and the oppsy module itself.

This is a really small data structure, so the time spent copying this is trivial.

I don't think consumers would expect that this data would change after they've gotten the event. And in fact, I think it'd be very common for this event to be used in an asynchronous operation. For any consumers using this module to track stats, it's common practice to buffer or rollup stats before flushing them to a data store.

I think the default behavior here is unexpected and does not follow typical Node conventions.

Reply:
I beg to differ, it's very common in node to emit objects that might be changed by subscribers, it's your role as a subscriber to clone it if you intend to modify it, the emitter is not supposed to know. Also, here you're cloning once, if there are several subscribers to the `ops` event, you'll still run into possible race conditions.

Reply:
That's a good point. I'll go ahead and close this PR.